### ISSUE TITLE
Property value has to be an instance of "Mongo.Collection"

### ISSUE THREAD

**Issue by dnish** [jagi/meteor-astronomy#656]
Hey,
after updating to the latest Meteor version (and packages), I get the following error message:

```
TypeError: ["Test" class]["collection" property] Property value has to be an instance of "Mongo.Collection"
```

This is how I've defined my collection:

```
const Tests = new Meteor.Collection("tests");

const Test = Class.create({
    name: 'Test',
    collection: Tests,
    fields: {
        name: {
            type: String
        }
    }
});
```

**Comment by dnish** [comment]
Just recognized that it has to be <code>new Mongo.Collection</code> since the latest update.

**Comment by lukejagodzinski** [comment]
You've probably been using very old version of Astronomy. Meteor changed the way how we define collections https://docs.meteor.com/api/collections.html but for backward compatibility `Meteor.Collection` still works. Moreover `Meteor.Collection === Mongo.Collection` so it shouldn't throw an error even if you use `Meteor.Collection`, so the problem was probably somewhere else and just by accident you've fixed it by using `Mongo.` instead of `Meteor.`. Closing this issue

**Comment by diavrank** [comment]
I'm having the same problem with Meteor.roles (collection from **meteor-roles** package). This is my code:

```ts
import { Meteor } from 'meteor/meteor';
import { Class } from 'meteor/jagi:astronomy';
import { Mongo } from 'meteor/mongo';
import { Roles } from 'meteor/alanning:roles';

export interface RoleType extends Roles.Role{
	_id: string,
	children: Array<Object>,
	publicName: string
}

export const Permission = Class.create<RoleType>({
	name: 'Role',
	collection: Meteor.roles as unknown as Mongo.Collection<RoleType>,
	fields: {
		children: [Object],
		publicName: String,
		name: String
	}
});
```

And I get this error:

```ts
W20210930-00:51:21.921(-5)? (STDERR) /Users/diavrank/.meteor/packages/meteor-tool/.2.4.0_1.dh083o.ogxoi++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20210930-00:51:21.922(-5)? (STDERR) 						throw(ex);
W20210930-00:51:21.923(-5)? (STDERR) 						^
W20210930-00:51:21.923(-5)? (STDERR) 
W20210930-00:51:21.923(-5)? (STDERR) TypeError: ["Role" class]["collection" property] Property value has to be an instance of "Mongo.Collection"
W20210930-00:51:21.924(-5)? (STDERR)     at throwParseError (packages/jagi:astronomy/lib/modules/core/utils/throw_parse_error.js:18:9)
W20210930-00:51:21.924(-5)? (STDERR)     at Module.onParseDefinition (packages/jagi:astronomy/lib/modules/storage/hooks/parse_definition.js:9:7)
W20210930-00:51:21.924(-5)? (STDERR)     at packages/jagi:astronomy/lib/core/class.js:234:14
W20210930-00:51:21.924(-5)? (STDERR)     at packages/jagi:astronomy/lib/core/module.js:106:7
W20210930-00:51:21.925(-5)? (STDERR)     at arrayEach (/Users/diavrank/.meteor/packages/jagi_astronomy/.2.7.3.1hymdr8.c2t7l++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/lodash/_arrayEach.js:15:9)
W20210930-00:51:21.925(-5)? (STDERR)     at forEach (/Users/diavrank/.meteor/packages/jagi_astronomy/.2.7.3.1hymdr8.c2t7l++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/lodash/forEach.js:38:10)
W20210930-00:51:21.925(-5)? (STDERR)     at Function.forEach (packages/jagi:astronomy/lib/core/module.js:105:5)
W20210930-00:51:21.926(-5)? (STDERR)     at Function.extend (packages/jagi:astronomy/lib/core/class.js:219:12)
W20210930-00:51:21.926(-5)? (STDERR)     at Function.create (packages/jagi:astronomy/lib/core/class.js:149:11)
W20210930-00:51:21.926(-5)? (STDERR)     at module (imports/api/Permissions/Permission.ts:12:33)
W20210930-00:51:21.926(-5)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:336:7)
W20210930-00:51:21.927(-5)? (STDERR)     at Module.require (packages/modules-runtime.js:238:14)
W20210930-00:51:21.927(-5)? (STDERR)     at Module.moduleLink [as link] (/Users/diavrank/.meteor/packages/modules/.0.17.0.1lcb376.ty5k++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/reify/lib/runtime/index.js:52:22)
W20210930-00:51:21.927(-5)? (STDERR)     at module (imports/api/Profiles/Profile.ts:1:38)
W20210930-00:51:21.928(-5)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:336:7)
W20210930-00:51:21.928(-5)? (STDERR)     at Module.require (packages/modules-runtime.js:238:14)
=> Exited with code: 1
=> Your application is crashing. Waiting for file change.

```

It is weird because with Meteor.users I didn't have that problem.

**Comment by lukejagodzinski** [comment]
@diavrank I guess `Meteor.roles` is not instance of `Mongo.Collection`. I've checked in the code https://github.com/Meteor-Community-Packages/meteor-roles/blob/master/roles/roles_common.js#L28 and it looks like it is and it's expose in both client and server. Just console.log `Meteor.roles` before creating model definition, maybe it's `undefined` for some reason.

**Comment by diavrank** [comment]
Hi, I did a console.log over `Meteor.roles` and it is an object of `Mongo.Collection` in the server console, however, maybe it is different from the client side because I've just realized that I also used that file in the **client-side** since I use the **RoleType** interface, so maybe that's the cause of this error. So, I guess this can be fixed by adding a `Meteor.isServer` flag in my code file. Thank you for the suggestion.
